scripts/giphy.py

```diff
@@ -0,0 +1,55 @@
+"""
+Respond to:
+
+    /giphy "phrase"
+
+with a random image from giphy based on that phrase.
+"""
+
+import asyncio
+import functools
+import hangups
+import logging
+import re
+import requests
+import urllib
+
+
+logger = logging.getLogger('hangbot.scripts.giphy')
+
+
+API_KEY = 'dc6zaTOxFJmzC'  # Use the public API key for now
+API_URL = 'http://api.giphy.com/v1/gifs/random'
+COMMAND = '/giphy '
+MESSAGE_REGEX = re.compile(r'^{}'.format(COMMAND), re.IGNORECASE)
+
+
+async def process_message(client, conversation, message_event):
+    if MESSAGE_REGEX.match(message_event.text):
+
+        params = {
+            'api_key': API_KEY,
+            'tag': message_event.text[len(COMMAND):]
+        }
+        url = '{}?{}'.format(API_URL, urllib.parse.urlencode(params))
```

Comment:
Optional: [you can use the `params` keyword argument in `requests.get` instead of encoding the parameters yourself](http://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls).